matsim/src/main/java/org/matsim/utils/objectattributes/attributeconverters/StringConverter.java

```diff
@@ -29,7 +30,7 @@
  * @author mrieser
  */
 public class StringConverter implements AttributeConverter<String> {
-	private final Map<String, String> stringCache = new HashMap<String, String>(1000);
+	private final Map<String, String> stringCache = new ConcurrentHashMap<>(1000);
 	@Override
 	public String convert(String value) {
 		String s = this.stringCache.get(value);
```

Comment:
With `ConcurrentHashMap` you also have to change the insertion logic, as you might have a race condition here as well, with two processes getting a `null` value at the same time and then updating the cache. You could use 
```
return map.computeIfAbsent(key, k -> k);
```